src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -0,0 +1,214 @@
+package io.strimzi.test.container;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import com.github.dockerjava.api.model.ContainerNetwork;
+import io.strimzi.utils.AuxiliaryVariables;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.images.builder.Transferable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * StrimziKafkaContainer is a single-node instance of Kafka using the image from quay.io/strimzi/kafka with the
+ * given version. There are two options for how to use it. The first one is using an embedded zookeeper which will run
+ * inside Kafka container. The Another option is to use @StrimziZookeeperContainer as an external Zookeeper.
+ * The additional configuration for Kafka broker can be injected via constructor. This container is a good fit for
+ * integration testing but for more hardcore testing we suggest using @StrimziKafkaCluster.
+ */
+public class StrimziKafkaContainer extends GenericContainer<StrimziKafkaContainer> {
+
+    private static final Logger LOGGER = LogManager.getLogger(StrimziKafkaContainer.class);
+
+    public static final int KAFKA_PORT = 9092;
+    public static final int ZOOKEEPER_PORT = 2181;
+
+    private static final String STARTER_SCRIPT = "/testcontainers_start.sh";
+    private static final String LATEST_KAFKA_VERSION;
+    private static final List<String> SUPPORTED_KAFKA_VERSIONS = new ArrayList<>(5);
+
+    private Map<String, String> kafkaConfigurationMap;
+    private String externalZookeeperConnect;
+    private int kafkaExposedPort;
+    private int brokerId;
+
+    static {
+        // Reads the supported_kafka.versions for the supported Kafka versions
+        InputStream kafkaVersionsInputStream = StrimziKafkaContainer.class.getResourceAsStream("/supported_kafka.versions");
+
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(kafkaVersionsInputStream, StandardCharsets.UTF_8))) {
+            String kafkaVersion;
+            while ((kafkaVersion = bufferedReader.readLine()) != null) {
+                SUPPORTED_KAFKA_VERSIONS.add(kafkaVersion);
+            }
+        } catch (IOException e) {
+            String message = "Unexpected IOException there is no `supported_kafka.versions` file!";
+            LOGGER.error(message, e);
+
+            throw new RuntimeException(message, e);
+        }
+
+        LOGGER.info("Supported Kafka versions: {}", SUPPORTED_KAFKA_VERSIONS);
+
+        // sort kafka version from low to high
+        Collections.sort(SUPPORTED_KAFKA_VERSIONS);
```

Comment:
You can't use lexicographic ordering on version numbers because it will treat 10.0 < 3.0. Of course, this won't be a problem until there _is_ a 10.0, but still...